core/common/src/files/Paths.kt

```diff
@@ -14,20 +14,107 @@ import kotlinx.io.Source
  *
  * **This API is unstable and subject to change.**
  */
-public expect class Path
+public expect class Path {
+    /**
+     * Returns a path representing a parent directory for this path,
+     * or `null` if there is no parent directory for this path.
+     *
+     * How the parent path is resolved is platform-specific.
+     */
+    public val parent: Path?
+
+    /**
+     * Returns the name of the file or directory represented by this path.
+     *
+     * How the name is resolved is platform-specific.
+     *
+     * In general, one may expect that for a path like `Path("home", "Downloads", "file.txt")`
+     * the name is `file.txt`.
+     */
+    public val name: String
+
+    /**
+     * Returns `true` if this path is absolute, `false` otherwise.
+     *
+     * How an absolute path is resolved is platform-specific.
+     */
+    public val isAbsolute: Boolean
+
+    /**
+     * Returns a string representation of this path.
+     *
+     * Note that the returned value will represent the same path as the value
+     * passed to [Path], but it may not be identical to it.
+     */
+    override fun toString(): String
+
+    override fun hashCode(): Int
+
+    /**
+     * Compares two paths for equality using its string representation ([toString]).
+     */
+    override fun equals(other: Any?): Boolean
+
+    public companion object {
+        /**
+         * A platform-specific character separating parts of the path.
+         *
+         * For example, it's usually `/` on Unix and `\` on Windows.
+         */
+        public val separator: Char
```

Comment:
Currently, Paths are paths in a default system's filesystem. It's unclear how to treat paths for different filesystems uniformly.

I can move the separator to FileSystem, but then some operations on Path, like `get parent` should be moved there too.

Reply:
Perhaps it is worth explicitly specifying that only file systems of the same type are supported within one instance of the application?

Reply:
I mentioned it in the kdoc, but it may be worth moving than property to a global scope (along w/ https://github.com/Kotlin/kotlinx-io/pull/213#discussion_r1321672604) and renaming it to a `SystemPathSeparator`.

Reply:
Done